src/codegen/compile.ml

```diff
@@ -6531,14 +6536,6 @@ module MakeSerialization (Strm : Stream) = struct
     G.i (GlobalGet (nr (E.get_global env "__typtbl_idltyps")))
 
   module Registers = struct
-    let register_globals env =
```

Comment:
Just an idea (of lowest importance, please feel free to ignore): In light of merging (and structure), we could maybe keep and extend this function at this place and pass the initial metering limit parameters as (structured) argument. The lower metering initialization logic could then call this register declaration function (e.g. with a name like `initialize`), while the module initialization calls again this more general initializer.

Reply:
Ok, I did something slight different that avoids moving register_globals elsewhere...